### ISSUE TITLE
RPG terms for new stats causing frustration for non-RPG users

### ISSUE THREAD

**Issue by ghost** [HabitRPG/habitica#2071]
Note: This got really long and I apologize for that.

The new classes feature has made HabitRPG users much more reliant on knowledge of RPG terminology. This is leaving us non-RPG users in the dust and causing unnecessary frustration. The main issue is that the new stats (PER, STR, CON, etc) are opaque to the non-RPG user. The relationship between task attributes (Physical, Emotional, Social, Other), gear function, and point allocation now hinge on understanding PER, STR, CON, etc. That makes it unclear how HP and XP (and now MP) are affected.

For example, the previous gear descriptions had mouseovers that stated in plain English how a particular piece of gear affected HP or XP. Current mouseovers require the user to have a working definition of PER, STR, CON, INT and if they don't...well, that leaves the user stuck and having to search around for definitions (and then having to remember them which is harder with no RPG experience to fall back on to make it intuitive).

Tweaking the mouseover in various places of the interface would make this clearer and help us non-RPGers out. 

For example (these all come from the Rogue class as it's what I have access to):

Gear:

The current description for Ninja-to is: "Sleek and deadly as the ninja themselves. Increases STR by 8."

A more informative description for the uninitiated might be a combination of the old and new description methods along the lines of: "Sleek and deadly as the ninja themselves. Increases XP gain by X% by increasing STR by 8."

Potions:

The Backstab spell currently reads: "Without a sound, you sweep behind a task and stab it in the back. You deal higher damage to the task, with a higher chance of a critical hit."

I had no idea what that meant until I used it since I don't know what a critical hit is. I suggest something along the lines of: "Without a sound, you sweep behind a task and stab it in the back. You deal higher damage to the task, with a higher chance of a critical hit, netting you more XP points."

The Pickpocket, Stealth, and Tools of the Trade spell definitions are clear as is, telling the user exactly what the effect is.

Task Allocation:

Here a mouseover with a short note about how each attribute relates to stats and how those relate to HP/XP/MP would help. For example:

Physical: Physical tasks increase STR which increase XP gain.

Stats & Achievements:

It would be really helpful if those mouseovers for STR, CON, PER, INT appeared as short definitions under the stats table (not in between each stat, but below and grouped together). The problem with mouseovers is that it is difficult to hold in your head what each one does as you are trying to decide where to allocate points if you have to mouseover each time. It looks like there is room (unless there are plans for that space in which case maybe a collapsible area underneath might work?).

Also, the definition for STR and CON are a little unclear to me: 

STR: "Strength increases damage you inflict to tasks (reducing redness), and increases critical hit chance."

What trips me up is "increases damage you inflict to tasks (reducing redness)." Is that more XP? Also "critical hit" needs a helper phrase for those unfamiliar with the term.

For CON: "buff spells" needs a helper phrase (I had to google it to figure out what it was).

These are all small things but I think they would reduce frustration.

**Comment by wc8** [comment]
I heartily agree with clarifying this more. In the meantime, I’m adding some descriptions to the wiki to try to help, for example, [Critical Hit](http://habitrpg.wikia.com/wiki/Critical_hit)

**Comment by JennaMalkin** [comment]
I'd love to see the mouseovers expanded as a small table below, so that all of them can be read at the same time and compared - often it's hard to get the mouse just right over the mouseover, and it becomes half-faded in, half-faded out.  

Strength doesn't impact XP or HP at all - it simply returns the tasks to Bright Blue more quickly.  (If you are Strong, clicking a Dark Red turns it Orange, not Red.) http://habitrpg.wikia.com/wiki/Strength

I'd love to see 'buff spells' explained in-site as well.

**Comment by wc8** [comment]
I like that idea for a permanent description beneath Attributes.

**Comment by wc8** [comment]
Oh, and I apologize for CON to whatever degree I added to it, I was originally going to vote for DEF and I changed it. 
Maybe we can come up with a short list of better descriptions, then this might even qualify for the Entry Level Coding/Wording label. If so, even a less experienced developer can turn this into a pull request to be reviewed and merged to the website.

**Comment by SabreCat** [comment]
Descriptions and popover text and such are totally entry-level coder material!

**Comment by ghost** [comment]
@JennaMalkin Heh, I knew I was going to go wrong somewhere in my examples. Which I guess proves my point that us non-RPGers are rather clueless on these terms. I looked for Strength in the glossary this morning but I guess I was looking in the wrong spot in the wiki (not used to navigating it yet).

**Comment by JennaMalkin** [comment]
That's why I always use the Search bar in the wiki, @Anysia. :) I've added all 4 attributes to the glossary, with their abbreviations.

**Comment by wc8** [comment]
I’m not an RPG expert, but still, while the terms made sense RPG wise to me, most of the implications in habitrpg were/are obscure for me.

**Comment by wc8** [comment]
@Anysia what do you think of this definition of Mana:
Mana (MP) Magical energy used to cast spells to aid motivation in special circumstances. Mana regenerates 10 points every day.

**Comment by wc8** [comment]
Suggestions: 
1. Link to Glossary in Tavern Resources :+1: 
2. Less likely, but maybe: Can the site draw from short wiki definition articles for mouseovers? 
For example, in the wiki pages we can use code like {{:Critical Hits}} to include wiki Then wiki editors could help keep descriptions clear. We could restrict editing on those pages if needed.

**Comment by ghost** [comment]
@wc8 For the Mana definition: "aid motivation in special circumstances" left me with two questions (1) how do they aid motivation and (2) what special circumstances (having used them I know the answers, but coming at it cold I would be wondering). I'd be inclined to go for something more utilitarian like "Mana points are magical energy used to cast spells that affect tasks/gameplay/something like that. Mana regenerates by 10 points every day." I realize that doesn't really fit with the flavour of the other descriptions and needs adjusting I just don't have the mental energy to contribute at the moment.

As for your other suggestion, I would really favour the second one. The objection to the glossary ONLY (although the glossary IN ADDITION would be great under Resources) is that it is much easier to 'play' HabitRPG and learn the terms when they are presented in context like with mouseovers rather than a separate list to be consulted. Having them pulled from the wiki would ensure consistency so that sounds like a really good idea if it's workable.

**Comment by wc8** [comment]
How about: "Magic power used to cast spells to affect tasks/gameplay for help with habit-building. Mana regenerates by 10 points every day.”
Ideally each spell would explain something about how it can be helpful for habit-building. My 2¢

I agree about it being to easier to play with good explanations in context, but where they aren’t clear, having a glossary could help. In fact, our wiki founder already put in a  pull request for it #2077.  

(2) would be quite nifty and efficient, but I don’t know if it is feasible. I’m asking though—nothing ventured nothing gained.

This issue is ending up kind of long without clear next actions for developers...

**Comment by ghost** [comment]
That definition works for me.

Next actions I can see are:
1) Changing existing unclear mouse over text
2) Putting a table of stat definitions under the point allocation list instead of using mouse overs (there seemed to be support for this)
3) Adding mouse over to Physical, Emotional, Social, Other in the tasks to explain how they are related to Str, per, con, Int and how that in turn relates to XP/HP/MP (not sure how much support thereis for this)
4) Linking the glossary on the Tavern references list
5) Determining the feasibility of pulling definitions from the wiki

I had intended to help with at least writing new definitions/mouse overs but now won't be able to do much until the new year for unforeseen reasons. Other than that I'm not sure what else can be done.

**Comment by DanielTheBard** [comment]
This kind of makes me think that habitrpg itself should have like a little glossary of terms on it? I don't think users should feel forced to go searching the wiki (which they might not even know exists) if they don't understand a term. They should be able to just look it up on the site itself. Because now that I look at it, critical hit isn't defined anywhere is it? 

However, as a gamer, I also recognize this a bit as people's first experiences with a game. They don't understand a lot of what's going on, but with a good game, they should be able to learn what things mean and do, as they play. It shouldn't require the need to beat them with explanations of everything. That said, most rpgs and strategy games have a glossary or resources/tutorials in game that the user can go back to at any time and reference if they don't understand something. 

I don't think the new system is TOO hard to understand. I mean, @Anysia, as you did things you started to understand what they meant, right? You just didn't understand at first, but you figured it out as you went along. I think non-gamers not grasping everything right away is to be expected. And it isn't a bad thing! :) We'd have to spoon-feed the user to avoid it, which might insult or annoy some users. I think what we should do though, is we should have a tab on the site with a list of the tutorials that they can redo at anytime they please. Along with a glossary of RPG terms from the site, explained better for non-gamers, and it'd explain better how the terms correlate to the site.

I actually think I should create a trello card for this, what do you think @wc8 ? 

Also, I actually glossed over the last few posts and @Anysia, noted you were kind of saying what I'm suggesting. :) Though, I'm suggesting we need an outright help tab on the site. Users shouldn't have to leave the main site to understand it.

EDIT: I just noticed we have a FAQ link at the bottom of the page that links to the wiki. I've literally never noticed this before. Ever. I'm not sure if that says something about this. I still think though that there should be a help section more apparent on the actual site. Obviously, there's a lot more that the wiki can help with that the site probably can't, but I think glossary+tutorials. I'll throw up a trello card anyway.

EDIT2: It's like 5:30am and I haven't slept so maybe this will explain my idea better than my ramble aha. https://trello.com/c/vtmq7AeQ/353-help-tab-tutorials-glossary-etc

EDIT3: I do think quite a bit of the mouseovers should explain things better though yes. #maybeisaidthat #proofreadingsforsquares

**Comment by bobbyroberts99** [comment]
I agree.

Daniel the Bard wrote:

**Comment by ghost** [comment]
I agree that going off the site, even to an associated wiki, should not be necessary. The sticking point for me is that there is a difference between the amount of effort I will put into figuring out a game (which comes out of my free time) and the amount of time I will put into figuring out a productivity tool (which comes out of my productive time). Since the update, it feels like it is too out of balance. I'm much more in favour of tweaking mouseover text and adding a few more (it would only take a few) than relying on a help tab (which is not a bad idea but it shouldn't be the first line of defence so to speak).

@DanielTheBard said "However, as a gamer, I also recognize this a bit as people's first experiences with a game. They don't understand a lot of what's going on, but with a good game, they should be able to learn what things mean and do, as they play."

The game isn't difficult, but it IS difficult to intuit much of the new system (in part because some definitions/explanations need tweaking or adding so that they tell the user more without an RPG background).

As far as figuring it out just by playing, the game doesn't give you enough to go on in terms of notifications to figure out the effects of critical hit and buff. There is no notification of a critical hit so there is no way for a user to make the connection (I think that has been brought up elsewhere and is on the list to be fixed already so I didn't include it before). SabreCat had to tell me that it was a critical hit, not a bug, and I wasn't the only one in the Tavern who was confused at the occasional high score.

**Comment by JennaMalkin** [comment]
What happens when a low-level user learns a spell?  How about, when a player learns their first buff spell, Justin comes and explains 

'You've just learned a buff spell, which you can cast to help your party with a temporary boost to their Strength, Constitution, Perception or Intelligence.  The effects of a buff spell will last until each party member's day finishes.'

**Comment by ghost** [comment]
I like that idea. It helps out new users without forever bugging more savvy users as they use the game.

**Comment by octern** [comment]
Quick game-design comment: In a good game, players should be able to learn how things work as they play. But HRPG is an exception to that rule, for a few reasons:

1) Many mechanics are probabilistic, AND you don't check off many things per day. If a PER increase takes you from 2 drops per day to 5 drops every 2 days (with substantial random deviations in both directions), that's going to be hard to notice. 

2) In the case of defense, there's a long lag between what you do and what happens; a lot of things are deliberately hidden from the player; and you rarely have the exact same thing happen twice. In a normal RPG, you get hit by enemies dozens of times, then you buy new armor and immediately get hit dozens more times and you can see that the numbers are smaller. In HRPG, you buy new armor, wait up to 24 hours, and then get one hit that adds together the unique combination of tasks you missed that day, including their never-quite-the-same color status. 

3) The organic difficulty / damage / reward system hides a lot of the calculations that would show the effects of your changes. Even after playing for months, I'm not sure exactly how much less red I should expect a red habit to become when I click on it, and I'd have trouble noticing how much a stat change altered that. 

I don't think of these as weaknesses in HRPG; rather, they're ways the game has taken on the substantial challenge of mapping RPG mechanics onto a world that runs much more slowly than the standard game, is less constrained, and offers fewer opportunities for built-in feedback. I think this is a good reason to make descriptions of the game mechanics very explicit, even if it does mean having lots of possbily-extraneous help text.

**Comment by quin-jay** [comment]
As far as the concern about annoying RPG veterans, is there any way we might be able to let a user switch back and forth between CSS versions? I'm picturing a checkbox in the Misc section of the Settings tab saying something along the lines of:

**Standard English Tooltips.** New to the land of RPGs? Choose this setting to change the game jargon around the site to more intuitive phrasings.

Then there would be some code somewhere like "if true then print {newbieCSS}"

This would be annoying to developers for two reasons:
1) Sounds like a lot of work (but the alternative is changing everything anyway, so if it's not too much harder to do this then why not)
2) Pushing changes across the site might require changing the code in twice as many places now (but maybe there's some way to only change the TEXT instead of the design?)

**Comment by Alys** [comment]
I'm closing this because since it was reported we've made many changes to the site, including significant amounts of help text for new users. The staff are also regularly monitoring anonymous usage patterns and use that information to tailor the site's behaviour and documentation to increase retention of new users (i.e., make the site easier to use for new users).

I think the best way now to handle confusing features or wording is to report each item individually in its own issue (or report a few related items together). We'll analyse each issue as it's reported.
